# Parameter for MPC solver
dT: 0.25 #unit: s
mpc_steps_N: 15
vehicle_L: 0.325 # distance between the front of the vehicle and its center of gravity

max_speed: 2.0 # unit: m/s
mpc_ref_cte: 0.0
mpc_ref_epsi: 0.0
mpc_ref_vel: 1.5

mpc_w_cte: 900
mpc_w_lag: 500
mpc_w_epsi: 0
mpc_w_vel: 3.0
mpc_w_p: 5.0
mpc_w_delta: 20
mpc_w_accel: 10
mpc_w_delta_d: 700
mpc_w_delta_p: 0

mpc_max_steering: 0.42 # Maximal steering radian (~30 deg)
mpc_max_throttle: 1.0 # Maximal throttle accel
mpc_bound_value: 1.0e3 # Bound value for other variables

p_max: 2.5
d_v_bound: 2.0
d_theta_bound: 0.5
d_p_bound: 2.0

# Parameters for control loop
debug_mode: True
delay_mode: True
throttle_mode: False
car_width: 0.3
spline_poly_order: 3
goal_threshold: 0.6 # unit: m
controller_freq: 15
path_folder_name: dearborn211_demo_track
inflation_factor: 0.2
lag_time: 0.15
ipopt_verbose: True
#Frame names
car_frame: base_link

#Topic names
odom_topic_name: /odom
goal_topic_name: /move_base_simple/goal
cmd_vel_topic_name: /nav
localized_pose_topic_name: /pf/viz/inferred_pose

